textile/features.textile

```diff
@@ -1655,6 +1656,17 @@ h4. DevicePushDetails
 * @(PCP3)@ recipient - a map of string key/value pairs containing details of the push transport and address
 * @(PCP4)@ state - the state of the push registration, one of  @Active@, @Failing@, @Failed@
 
+h3(#introspection). Client library introspection
+
+h4. ClientLibraryInfo
+* @(CLI1)@ Provides information about the client library.
+* @(CLI2)@ @agents@ static property:
+** @(CLI2a)@ Returns a @Dict<String, String?>@ that lists the default key-value entries that the library uses to populate the @Agent@ library identifier, as described by "@RSC7d@":#RSC7d.
+* @(CLI3)@ @agentIdentifier@ static method:
+** @(CLI3a)@ Returns the @Agent@ library identifier described by "@RSC7d@":#RSC7d.
+** @(CLI3b)@ If the library offers the @ClientOptions#agents@ property described by "@RSC7d6@":#RSC7d6, then the @agentIdentifier@ method must accept an @additionalAgents@ parameter of type @Dict<String, String?>?@, whose value is to be interpreted with the same semantics as the @ClientOptions#agents@ property. The @agentIdentifier@ method will inject these agents into the @Agent@ library identifier that it returns.
```

Comment:
This spec point is hard to prove downstream in SDKs, due to variability of the interface shape. Also, when an SDK then goes on to implement `RSC7d6` after having only previously implemented `CLI3b` then we're going to have to make a breaking change just to refactor this method. That feels like planned maintenance headache to me.

Might it not just be clearer to state that this method should only be implemented if `RSC7d6` has _also_ been implemented? Then it will always have the same signature.

Reply:
I've changed it to say that "The client library must offer this method if and only if it offers the ClientOptions#agents property described by RSC7d6".

Reply:
Thanks, though I think it needs further refinement. Hence: https://github.com/ably/specification/pull/113#pullrequestreview-1182941317